### DISCUSSION TITLE
Improve PrettyBlocks

### DISCUSSION THREAD

**Discussion by LuckyMarty** [PrestaSafe/prettyblocks#143] [Ideas]
Hello everyone!

Currently, when I have a multilingual site, it is not possible to modify a text according to the language.
Maybe add, as on the modules, a small list of languages ... ?

Another suggestion is that when I create the field that allows me to add an image, I get the window that opens and by default it authorizes all extensions.
Is it possible to add a parameter that lets you enter the possible extensions to be uploaded?

**Comment by PrestaSafe** [comment]
Hi 
For the language, you need to go on the language inside the prettyblocks iframe, and the blocks will be loaded by using the current website language. 

Fot the upload, we will see how to handle this easily, we have change that recently for adding document like pdf, what kind of file you want to add ?

**Reply by LuckyMarty** [reply]
Hi!

Woaw 😮
Thank you for your quick response.

I was so into the Prestashop method that I didn't think that way.

For the upload extenssion I was thinking like, when I would like to upload an image, only image extensions are allowed.
```PHP
'upload' => [
    'type' => 'fileupload', // type of field
    'label' => 'File upload', // label to display
    'path' => '$/cz_gallery/imgs/', // path to upload
    'default' => [ // default value
        ['url' => 'https://via.placeholder.com/100x100'],
    ],
    'extenssion' => ['.png', '.jpg', '.webp'] // default ALL
]
```

**Reply by PrestaSafe** [reply]
I will put a flag for the language, because lot of people do not understand that, it's maybe too simple ? :) For extensions, i will see that. 
Thanks.

**Comment by LuckyMarty** [comment]
Bonjour !

Je suis de retour en français 😅

J'ai pu suivre quelques de vos vidéos et j'ai réussi à intégrer une partie de ma fonctionnalité.
Par contre je n'ai pas la possibilité de PR.


------------------------

**Qu'est-ce qui a été ajouté :**
La possibilité de filtrer les fichiers que l'on souhaite uploader

**Comment cela fonctionne ?**
- Soit en utilisant le MIME (image/*)
`'fileMime' => 'image/webp,image/png',`

- Soit en utilisant directement l'extension du fichier
`'fileExtension' => '.jpeg,.png,.gif,.webp',`

**Si deux méthodes sont utilisées :**
`'fileMime' => 'image/webp,image/png',`
`'fileExtension' => '.jpeg,.png,.gif,.webp',`
La méthode 'fileMime' sera prioritaire

**Problématique :**
Pour l'instant je n'ai que réussit à faire fonctionner dans les blocs répétitifs ...
Comment fait-on pour gérer les props pour les blocs simples ?


```
'repeater' => [
	'name' => 'Slides',
	'nameFrom' => 'title',
	'groups' => [
		'upload' => [
			'type' => 'fileupload',
			'label' => 'Images',
			'path' => '$/modules/classicblocks/views/images/',
			'fileMime' => 'image/webp,image/png', // Cette ligne à la priorité sur "fileExtension"
			'fileExtension' => '.jpeg,.png,.gif,.webp', // Si vous souhaitez utiliser ce format, commentez la ligne "fileMime"
			'default' => [
				'url' => 'https://via.placeholder.com/1110x340',
			],
		]
	]
]
```

**Reply by PrestaSafe** [reply]
Bonjour, Pour la PR, tout le monde peut en faire, je ne vois pas où cela peut bloquer. 
Pour cette feature, cela implique de très lourd changements avec énormement de tests car niveau sécurité cela peut être très intrusif.
Avoir 2 possibilités d'upload semble un peu compliqué, je pense qu'il faut rester uniquement sur les extensions `fileExtension` 
Il faut également regarder au niveau du controller SF pour l'upload: https://github.com/PrestaSafe/prettyblocks/blob/f2fe6beff0a2d9041ba58c3217ee553ccf0e9f21/src/Controller/AdminThemeManagerController.php#L31

Cordialemet

**Reply by LuckyMarty** [reply]
Je vois ...

Pour l'instant je n'ai seulement fait coté client.
Je viens de créer une PR. Je vous laisse voir les quelques changement.
Cela reste une proposition :)

Bien à vous
